acer-wmi-battery.c

```diff
@@ -217,6 +253,28 @@ static void update_state(void)
 			battery_status.health_mode ? "enabled" : "disabled");
 }
 
+static ssize_t temperature_show(struct device_driver *driver, char *buf)
+{
+	acpi_status status;
+	u32 value;
+
+	/*
+	 * We always use battery number 0x1 to stay consistent with the
+	 * battery health mode interface.
+	 *
+	 * The information index 0x8 was taken from the
+	 * "Smart Battery Data Specification".
+	 */
+	status = get_battery_information(0x8, 0x1, &value);
+	if (ACPI_FAILURE(status))
+		return -EIO;
+
+	if (value > U16_MAX)
+		return -ENXIO;
+
+	return sysfs_emit(buf, "%u\n", (value - 2731) * 100);
```

Comment:
Is the expression guaranteed to be non-negative? Perhaps it should be printed as `%d` instead? Negative values seem like a possibility here, although I wouldn't expect them to occur often ;-).

Reply:
Ok, will fix.